spec/engines/search_engine_spec.rb

```diff
@@ -17,6 +17,9 @@ def parse_search_engine_response(_)
 end
 
 describe SearchEngine do
+  before(:all) do
+    CachedSearchApiConnectionResponse = Struct.new(:response, :cache_namespace)
```

Comment:
@HalahAb , I see the issue with this spec, but I don't think that redefining the struct in the spec is the best approach. I think a safer fix would be to move the struct definition to `lib/cached_search_api_connection_response.rb`, to ensure that that class is auto-loaded.